hack/dependencies.yml

```diff
@@ -1,16 +1,134 @@
----
-ytt_version: v0.40.1
-ytt_checksum_darwin: d46dba5e729e2fe36c369e96eaa2eb5354fb1bf7cf9184f9bfa829b8e5558b94
-ytt_checksum_linux: 11222665c627b8f0a1443534a3dde3c9b3aac08b322d28e91f0e011e3aeb7df5
-kbld_version: v0.33.0
-kbld_checksum_darwin: 181ac8be5652b54344617d90aa8e83fbb41756d1b4b99168fec85d8813b3c1b2
-kbld_checksum_linux: 38a5dad7ed478d209c8206d95546989b2730c7fed914c78d85eed68a2233688e
-kapp_version: v0.47.0
-kapp_checksum_darwin: ed0a7a86828517e7aa3d4bede659ac725806b0201692f2d574beb71785220406
-kapp_checksum_linux: 2b1bd5e3a35ed63045be80a7cc8cbc054fe4e98c89c55f9006c8fec2243658a4
-imgpkg_version: v0.28.0
-imgpkg_checksum_darwin: e43142fdb197a62844acb29cb619d513346aac3c23732a4d180c0ad974d9562e
-imgpkg_checksum_linux: 8d22423dd6d13efc0e580443d8f88d2183c52c6f851ba51e3e54f25bf140be58
-vendir_version: v0.27.0
-vendir_checksum_darwin: c26547097d67f21e129a25557d9d36c7c0e109afe130adff63d3c83ce9459ecc
-vendir_checksum_linux: 1aa12d070f2e91fcb0f4d138704c5061075b0821e6f943f5a39676d7a4709142
+- autoupdate:
+    checksums:
+      file: checksums.txt
+    github: vmware-tanzu/carvel-ytt
+  checksums:
+    darwin:
+      amd64: d46dba5e729e2fe36c369e96eaa2eb5354fb1bf7cf9184f9bfa829b8e5558b94
+      arm64: a478eb4a6a88a8e3de9dd2e97c1fb96ca7a65c74525f2e132e6a5b333803f7c8
+    linux:
+      amd64: 11222665c627b8f0a1443534a3dde3c9b3aac08b322d28e91f0e011e3aeb7df5
+      arm64: c5d2f033b375ee87414b51d16c324d7a441de2f256865b5c774c4d5aea30ef60
+  dev: true
```

Comment:
It feels like there should be a doc that describes (or comments) these fields without looking at the kinda sorta ugly go code. Like what does `dev: true` vs `dev: false` mean? And `prod`?

Reply:
I also don't THINK we have tests, so just wondering what happens if we mess up some of these field positions in yaml, does it fail silently?

Reply:
> without looking at the kinda sorta ugly go code.

I'd rather we make the code reasonable enough to understand rather than maintain documentation for what's essentially a fancy bash script

> I also don't THINK we have tests, so just wondering what happens if we mess up some of these field positions in yaml, does it fail silently?
Yeah there aren't tests, though there weren't tests for our Dockerfile/dependencies.yml before either. I guess whether it causes something to fail or not depends on what you change in the YAML file.

- If you made a `dev` dependency no longer `dev`, the tests/other scripts that depend on those tools being installed would fail.
- If you removed a checksum it would no longer install on that os/arch

Reply:
> I'd rather we make the code reasonable enough to understand rather than maintain documentation for what's essentially a fancy bash script

Fair, I was referring to the `dependencies.yml` structure though, not the code. The code can parse, ignore, whatever the fields in this yaml file. But how do I know what to put in here, without reading gocode? Why should I read gocode for an api? That's how I am reading this `dependencies.yml` file. It's our interface for dependencies.

Reply:
I did another pass at the code and removed a lot of the more confusing elements of the `dependencies.yml` now. I still would be ok with documenting the structure somehow if you think that'd be helpful.

Reply:
I think it's fine to leave as is. The fields are self-explanatory (I was just confused by the `dev` vs `prod` flags and how they interacted with each other). Looks like `prod` was removed, so 👍